js/components/reorderList.js

```diff
@@ -69,15 +71,35 @@ Fliplet.FormBuilder.field('reorderList', {
     }
   },
   watch: {
+    options: {
+      immediate: true,
+      handler(val) {
+        this.orderedOptions = _.cloneDeep(val);
+      }
+    },
     value: function(val) {
       if (val) {
-        this.orderedOptions = val.map(value => this.options.find(option => option.id === value));
+        const allValuesPresent = val.every(value =>
+          this.orderedOptions.some(option => option.id === value)
+        );
+
+        if (allValuesPresent && val.length === this.orderedOptions.length) {
+          // Create a map to store the indexes of options based on their IDs
+          const indexMap = new Map();
+
+          this.orderedOptions.forEach((option, index) => {
+            indexMap.set(option.id, index);
+          });
```

Comment:
[MINOR] @armine-fliplet you can utilize here the Map constructor

```suggestion
          const indexMap = new Map(this.orderedOptions.map((option, index) => [option.id, index]));
```

Reply:
Thanks @dwfliplet